psd-web/spec/models/user_spec.rb

```diff
@@ -417,4 +419,81 @@
       end
     end
   end
+
+  describe "devise gems echosystem" do
+    describe "#send_two_factor_authentication_code", :with_test_queue_adpater do
+      subject(:user) { create(:user) }
+
+      # rubocop:disable RSpec/MultipleExpectations, RSpec/ExampleLength
+      it "enqueues a SendTwoFactorAuthenticationJob" do
+        ActiveJob::Base.queue_adapter = :test
+        expect { user.send_new_otp }
+          .to enqueue_job(SendTwoFactorAuthenticationJob)
+                .at(:no_wait)
+                .on_queue("psd")
+                .with do |user_param, code_param|
+          expect(user_param).to be user
+          expect(user.direct_otp).to eq(code_param)
+        end
+      end
+      # rubocop:enable RSpec/MultipleExpectations, RSpec/ExampleLength
+    end
+
+    describe "#two_factor_lock_expired?" do
+      it "defaults to true when there was no two factor lock for the user" do
+        user = build_stubbed(:user, second_factor_attempts_locked_at: nil)
+        expect(user.two_factor_lock_expired?).to eq true
```

Comment:
consider using rspec syntactic sugar, it reads much nicer:
```suggestion
        expect(user).to be_two_factor_lock_expired
```